pyface/ui/wx/font.py

```diff
@@ -0,0 +1,183 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+"""
+Font conversion utilities
+
+This module provides facilities for converting between pyface Font objects
+and Wx Font objects, trying to keep as much similarity as possible between
+them.
+"""
+
+import wx
+
+from pyface.font import Font
+
+
+# font weight and size features changed in wxPython 4.1/wxWidgets 3.1
+wx_version = tuple(int(x) for x in wx.__version__.split('.')[:2])
+wx_python_4_1 = (wx_version >= (4, 1))
+
+
+wx_family_to_generic_family = {
+    wx.FONTFAMILY_DEFAULT: 'default',
+    wx.FONTFAMILY_DECORATIVE: 'fantasy',
+    wx.FONTFAMILY_ROMAN: 'serif',
+    wx.FONTFAMILY_SCRIPT: 'cursive',
+    wx.FONTFAMILY_SWISS: 'sans-serif',
+    wx.FONTFAMILY_MODERN: 'monospace',
+    wx.FONTFAMILY_TELETYPE: 'typewriter',
+}
+generic_family_to_wx_family = {
+    'default': wx.FONTFAMILY_DEFAULT,
+    'fantasy': wx.FONTFAMILY_DECORATIVE,
+    'decorative': wx.FONTFAMILY_DECORATIVE,
+    'serif': wx.FONTFAMILY_ROMAN,
+    'roman': wx.FONTFAMILY_ROMAN,
+    'cursive': wx.FONTFAMILY_SCRIPT,
+    'script': wx.FONTFAMILY_SCRIPT,
+    'sans-serif': wx.FONTFAMILY_SWISS,
+    'swiss': wx.FONTFAMILY_SWISS,
+    'monospace': wx.FONTFAMILY_MODERN,
+    'modern': wx.FONTFAMILY_MODERN,
+    'typewriter': wx.FONTFAMILY_TELETYPE,
+}
+
+if wx_python_4_1:
+    weight_to_wx_weight = {
+        100: wx.FONTWEIGHT_THIN,
+        200: wx.FONTWEIGHT_EXTRALIGHT,
+        300: wx.FONTWEIGHT_LIGHT,
+        400: wx.FONTWEIGHT_NORMAL,
+        500: wx.FONTWEIGHT_MEDIUM,
+        600: wx.FONTWEIGHT_SEMIBOLD,
+        700: wx.FONTWEIGHT_BOLD,
+        800: wx.FONTWEIGHT_EXTRABOLD,
+        900: wx.FONTWEIGHT_HEAVY,
+        1000: wx.FONTWEIGHT_EXTRAHEAVY,
+    }
+    wx_weight_to_weight = {
+        wx.FONTWEIGHT_THIN: 'thin',
+        wx.FONTWEIGHT_EXTRALIGHT: 'extra-light',
+        wx.FONTWEIGHT_LIGHT: 'light',
+        wx.FONTWEIGHT_NORMAL: 'normal',
+        wx.FONTWEIGHT_MEDIUM: 'medium',
+        wx.FONTWEIGHT_SEMIBOLD: 'semibold',
+        wx.FONTWEIGHT_BOLD: 'bold',
+        wx.FONTWEIGHT_EXTRABOLD: 'extra-bold',
+        wx.FONTWEIGHT_HEAVY: 'heavy',
+        wx.FONTWEIGHT_EXTRAHEAVY: 'extra-heavy',
+        wx.FONTWEIGHT_MAX: 'extra-heavy',
+    }
+else:
+    weight_to_wx_weight = {
```

Comment:
atm im unable to find an external reference for this.

Reply:
There isn't one - this is essentially arbitrary, but hopefully sensible.